File: payment-service/src/main/java/com/hedvig/paymentservice/domain/payments/commands/CreateChargeCommand.java
package com.hedvig.paymentservice.domain.payments.commands;

import lombok.Value;
import org.axonframework.commandhandling.TargetAggregateIdentifier;

import javax.money.MonetaryAmount;
import java.time.Instant;
import java.util.UUID;

@Value
public class CreateChargeCommand {
  @TargetAggregateIdentifier String memberId;

  UUID transactionId;
  MonetaryAmount amount;
  Instant timestamp;
  String email;
  String createdBy;
}

File: payment-service/src/main/java/com/hedvig/paymentservice/domain/payments/commands/CreateChargeCommand.kt
package com.hedvig.paymentservice.domain.payments.commands

import java.time.Instant
import java.util.UUID
import javax.money.MonetaryAmount
import org.axonframework.commandhandling.TargetAggregateIdentifier

data class CreateChargeCommand(
    @TargetAggregateIdentifier  
    val memberId: String,
    val transactionId: UUID,
    val amount: MonetaryAmount,
    val timestamp: Instant,
    val email: String,
    val createdBy: String
)
